src/aead.rs

```diff
@@ -148,13 +149,170 @@ pub fn open(
 	Ok(dst_out)
 }
 
+/// Stream based authenticated secret-key encryption.
+///
+/// # Use case:
+///  `StreamSealer` can be used to encrypt and authenticate a stream of data. It prevents the
+///  modification, reordering, dropping or duplicating of messages. Nonce management is handled automatically
+///
+///  An example of this could be the encryption of files which are too large to encrypt in one piece.
+///
+/// # About:
+///
+/// This implementation is based on and compatible with the ["secretstream" API](https://download.libsodium.org/doc/secret-key_cryptography/secretstream)
+/// of libsodium.
+///
+/// # Parameters:
+/// - `secret_key`: The secret key.
+/// - `nonce`: The nonce value.
+/// - `plaintext`: The data to be encrypted.
+/// - `ciphertext`: The encrypted data with a Poly1305 tag and a [`StreamTag`](./hazardous/aead/streaming/enum.StreamTag.html) indicating its function.
+/// - `tag`: Indicates the type of message. The `tag` is a part of the output when encrypting. It
+/// is encrypted and authenticated.
+///
+/// # Errors:
+/// An error will be returned if:
+/// - The length of `ciphertext` is not greater than `16`.
+/// - The received mac does not match the calculated mac when decrypting. This can indicate
+///   a dropped or reordered message within the stream.
+/// - More than 2^32-3 * 64 bytes of data are processed when encrypting/decrypting a single chunk.
```

Comment:
We should also document that an error is returned if `SecretKey` is not 32 bytes in length.